WordPress/Classes/Utility/iAds/SearchAdsAttribution.swift

```diff
@@ -0,0 +1,124 @@
+/// Implementation of Search Ads attribution details
+/// More info: https://searchads.apple.com/help/measure-results/#attribution-api
+
+import Foundation
+import iAd
+import AutomatticTracks
+
+@objc final class SearchAdsAttribution: NSObject {
+
+    /// Keep the instance alive
+    ///
+    private static var lifeToken: SearchAdsAttribution?
+
+    private static let userDefaultsSentKey = "search_ads_attribution_details_sent"
+    private static let userDefaultsLimitedAdTrackingKey = "search_ads_limited_tracking"
+
+    /// Is ad tracking limited?
+    /// If the user has limited ad tracking, and this API won't return data
+    ///
+    private var isTrackingLimited: Bool {
+        get {
+            return UserDefaults.standard.bool(forKey: SearchAdsAttribution.userDefaultsLimitedAdTrackingKey)
+        }
+        set {
+            UserDefaults.standard.set(newValue, forKey: SearchAdsAttribution.userDefaultsLimitedAdTrackingKey)
+        }
+    }
+
+    /// Has the attribution details been sent already?
+    /// Once the attribution details are sent, it won't change; so it is not necesary to send it again
+    ///
+    private var isAttributionDetailsSent: Bool {
+        get {
+            return UserDefaults.standard.bool(forKey: SearchAdsAttribution.userDefaultsSentKey)
+        }
+        set {
+            UserDefaults.standard.set(newValue, forKey: SearchAdsAttribution.userDefaultsSentKey)
```

Comment:
Missing `UserDefaults.standard.synchronize()`

Reply:
@jleandroperez 

I implemented it but then I saw in its description this:
` this method is unnecessary and shouldn't be used.` 
So I removed it.

[Here are the docs](https://developer.apple.com/documentation/foundation/userdefaults/1414005-synchronize)

Now I'm reading a bit more about it and it seems to be deprecated from iOS10.
You can see in the "expanded" docs (cmd+click in the method name) this interesting message:

> **-synchronize is deprecated and will be marked with the NS_DEPRECATED macro in a future release.**
>  
>  -synchronize blocks the calling thread until all in-progress set operations have completed. This is no longer necessary. Replacements for previous uses of -synchronize depend on what the intent of calling synchronize was. If you synchronized…
>  — …before reading in order to fetch updated values: remove the synchronize call
>  — …after writing in order to notify another program to read: the other program can use KVO to observe the default without needing to notify
>  — …before exiting in a non-app (command line tool, agent, or daemon) process: call CFPreferencesAppSynchronize(kCFPreferencesCurrentApplication)
>  — …for any other reason: remove the synchronize call

Reply:
Ahhhhhaa! definitely new to me! This used to be mandatory, and without it, data wouldn't have been saved to disk.

So many calls we can nuke now!! thanks for sharing!!